tensorflow/core/kernels/eigen_attention.h

```diff
@@ -144,8 +144,11 @@ struct GlimpseExtractionOp {
       slice_extent[2] = std::min<Index>(input_height, slice_extent[2]);
 
       if (partial_overlap) {
-        if (uniform_noise_) {
-          // Initialize the glimpse with uniform noise.
+        if (noise_ == "zero") {
+           // Initialize the glimpse with zero noise.
+          output.template chip<3>(i).setZero();
```

Comment:
Can you specify the device to preserve the GPU compatibility ? You can call output.template chip<3>(i).device().setZero();, or if that's not implemented something like output.template chip<3>(i) = output.template chip<3>(i).constant(0);

Reply:
Thanks @benoitsteiner. Updated.